### PR TITLE
CLOUD-789 - add test to check password leaks in the logs

### PR DESCRIPTION
**Pull request by ptankov**
[CLOUD-789](https://jira.percona.com/browse/CLOUD-789) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
*Short explanation of the problem.*

**Cause:**
*Short explanation of the root cause of the issue if applicable.*

**Solution:**
*Short explanation of the solution we are providing with this PR.*

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [ ] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [ ] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [ ] Is an E2E test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [ ] Are all needed new/changed options added to default YAML files?
- [ ] Are the manifests (crd/bundle) regenerated if needed?
- [ ] Did we add proper logging messages for operator actions?
- [ ] Did we ensure compatibility with the previous version or cluster upgrade process?
- [ ] Does the change support oldest and newest supported PG version?
- [ ] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

File: e2e-tests/tests/init-deploy/03-read-from-primary.yaml [hunk 1]
```diff
@@ -10,3 +10,5 @@ commands:
       data=$(run_psql_local '\c myapp \\\ SELECT * from myApp;' "postgres:$(get_psql_user_pass init-deploy-pguser-postgres)@$(get_psql_user_host init-deploy-pguser-postgres)")
 
       kubectl create configmap -n "${NAMESPACE}" 03-read-from-primary --from-literal=data="${data}"
+
+      check_passwords_leak
```
Comment (tplavcic) [inline e2e-tests/tests/init-deploy/03-read-from-primary.yaml:14]: please add it as a separate step `04-check-pass-leak.yaml`

Comment (tplavcic) [reply e2e-tests/tests/init-deploy/03-read-from-primary.yaml:14]: also add it to `monitoring` test in position `06-check-pass-leak.yaml` and to `demand-backup` test into position `08-check-pass-leak.yaml`

**Review by tplavcic — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/ddf617e80/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/ddf617e80/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/ddf617e80/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/ddf617e80/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/ddf617e80/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/ddf617e80/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/ddf617e80/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/ddf617e80/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/ddf617e80/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/ddf617e80/users.log) |
| We run 10 out of 10|

commit: https://github.com/percona/percona-postgresql-operator/pull/489/commits/ddf617e80665c83252574cfc49407b60141b5afd
image: `perconalab/percona-postgresql-operator:PR-489-ddf617e80`

File: e2e-tests/tests/init-deploy/03-read-from-primary.yaml (hunk 1 above)
Comment (ptankov) [reply e2e-tests/tests/init-deploy/03-read-from-primary.yaml]: done

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/d54914a60/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/d54914a60/init-deploy.log) |
| monitoring | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/d54914a60/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/d54914a60/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/d54914a60/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/d54914a60/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/d54914a60/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/d54914a60/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/d54914a60/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/d54914a60/users.log) |
| We run 10 out of 10|

commit: https://github.com/percona/percona-postgresql-operator/pull/489/commits/d54914a607d1a1829b95db126b649adc2e713d5a
image: `perconalab/percona-postgresql-operator:PR-489-d54914a60`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/bd392b5b5/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/bd392b5b5/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/bd392b5b5/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/bd392b5b5/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/bd392b5b5/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/bd392b5b5/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/bd392b5b5/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/bd392b5b5/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/bd392b5b5/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-489/bd392b5b5/users.log) |
| We run 10 out of 10|

commit: https://github.com/percona/percona-postgresql-operator/pull/489/commits/bd392b5b5dd3c5a100149b8c6a870b76f7da2f82
image: `perconalab/percona-postgresql-operator:PR-489-bd392b5b5`

**Review by tplavcic — APPROVED** [review]

**Review by nmarukovich — APPROVED** [review]

**Pull request merged**